tests/acceptance/01_vars/02_functions/string_replace.cf

```diff
@@ -0,0 +1,49 @@
+
+#######################################################
+#
+# Test string_replace function
+#
+#######################################################
+
+body common control
+{
+      inputs => { "../../default.cf.sub" };
+      bundlesequence  => { default("$(this.promise_filename)") };
+      version => "1.0";
+}
+
+#######################################################
+
+bundle agent test
+{
+  vars:
+      "test" string => "abcdefghij\t\n";
+      "test2" string => "(){}[].*?";
+
```

Comment:
I would add some **really** basic test cases here first. It's great that you're testing special characters and edge cases, but if someone new is reading this test it is easier if you start with the "normal usage".

```
	"match_once" string => string_replace("abcdef", "abc", "ABC");
    "match_twice" string => string_replace("abcdabcd", "abcd", "hello");
    "match_none" string => string_replace("abdc", "abcd", "wrong");
```

And then, of course, compare results.